Authenticator rejection results in MQTTMessageLogger error

### Expected behavior

Rejection of authentication should not result in errors.

### Actual behavior

Rejection of authentication results in an error in the MQTTMessageLogger and a failed `massage` ;-)

```
11:01:25.157 | ERROR | i.m.broker.MQTTConnection | Authenticator has rejected the MQTT credentials CId=30c86d37-8c3f-4e00-8d66-6b875a3a3345, username=invalidTokenUser
11:01:25.158 | DEBUG | i.m.b.m.MQTTMessageLogger | C<-B CONNACK <null>
11:01:25.166 | DEBUG | i.m.b.AutoFlushHandler | Initializing autoflush handler on channel [id: 0xc5a0a3e3, L:/10.0.75.1:1883 - R:/10.0.75.1:55012] Cid: null
11:01:25.169 | ERROR | i.m.b.NewNettyMQTTHandler | Unexpected exception while processing MQTT message. Closing Netty channel. CId=null
java.io.IOException: invalid massage
	at io.moquette.broker.NettyUtils.validateMessage(NettyUtils.java:86)
	at io.moquette.broker.metrics.MQTTMessageLogger.logMQTTMessage(MQTTMessageLogger.java:61)
	at io.moquette.broker.metrics.MQTTMessageLogger.logMQTTMessageRead(MQTTMessageLogger.java:50)
```

```
Caused by: io.netty.handler.codec.mqtt.MqttIdentifierRejectedException: invalid clientIdentifier: 30c86d37-8c3f-4e00-8d66-6b875a3a3345
	at io.netty.handler.codec.mqtt.MqttDecoder.decodeConnectionPayload(MqttDecoder.java:341)
	at io.netty.handler.codec.mqtt.MqttDecoder.decodePayload(MqttDecoder.java:313)
	at io.netty.handler.codec.mqtt.MqttDecoder.decode(MqttDecoder.java:100)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:366)
```

### Steps to reproduce

- Start a Moquette broker that requires authentication
- Connect with an MQTT Client using username/password (I used the paho Java client)
- Error occurs on the CONNECT -> CONNACK sequence

### Minimal yet complete reproducer code (or URL to code) or complete log file

I cant provide the code unfortunately. Reproducing should be simple though. If needed I can build a standalone example, but it will take me an hour or two. Here you can find a full log file:

https://pastebin.com/x9tNYVE0

### Moquette MQTT version

0.12.1

### JVM version (e.g. `java -version`)

Java 8 1.8.0_201

### OS version (e.g. `uname -a`)

Windows 10

--- a/broker/src/main/java/io/moquette/broker/MQTTConnection.java
+++ b/broker/src/main/java/io/moquette/broker/MQTTConnection.java
@@ -140,13 +140,13 @@ final class MQTTConnection {
         final boolean cleanSession = msg.variableHeader().isCleanSession();
         if (clientId == null || clientId.length() == 0) {
             if (!brokerConfig.isAllowZeroByteClientId()) {
-                LOG.warn("Broker doesn't permit MQTT empty client ID. Username: {}, channel: {}", username, channel);
+                LOG.info("Broker doesn't permit MQTT empty client ID. Username: {}, channel: {}", username, channel);
                 abortConnection(CONNECTION_REFUSED_IDENTIFIER_REJECTED);
                 return;
             }
 
             if (!cleanSession) {
-                LOG.warn("MQTT client ID cannot be empty for persistent session. Username: {}, channel: {}",
+                LOG.info("MQTT client ID cannot be empty for persistent session. Username: {}, channel: {}",
                          username, channel);
                 abortConnection(CONNECTION_REFUSED_IDENTIFIER_REJECTED);
                 return;
@@ -229,17 +229,17 @@ final class MQTTConnection {
             if (msg.variableHeader().hasPassword()) {
                 pwd = msg.payload().password().getBytes(StandardCharsets.UTF_8);
             } else if (!brokerConfig.isAllowAnonymous()) {
-                LOG.error("Client didn't supply any password and MQTT anonymous mode is disabled CId={}", clientId);
+                LOG.info("Client didn't supply any password and MQTT anonymous mode is disabled CId={}", clientId);
                 return false;
             }
             final String login = msg.payload().userName();
             if (!authenticator.checkValid(clientId, login, pwd)) {
-                LOG.error("Authenticator has rejected the MQTT credentials CId={}, username={}", clientId, login);
+                LOG.info("Authenticator has rejected the MQTT credentials CId={}, username={}", clientId, login);
                 return false;
             }
             NettyUtils.userName(channel, login);
         } else if (!brokerConfig.isAllowAnonymous()) {
-            LOG.error("Client didn't supply any credentials and MQTT anonymous mode is disabled. CId={}", clientId);
+            LOG.info("Client didn't supply any credentials and MQTT anonymous mode is disabled. CId={}", clientId);
             return false;
         }
         return true;

